### DISCUSSION TITLE
Using ref inside Controller render  causing 'Cannot set property value of # which has only a getter'

### DISCUSSION THREAD

**Discussion by abeprincec** [react-hook-form/react-hook-form#5968] [Q&A]
**Describe the bug**
When using ref prop inside Controller render attribute it causes the form to crash, 
The Component I am using accepts a ref prop

**To Reproduce**
Use the Kendo Date Picker with react-hook-form

**Codesandbox link (Required)**
V7
https://codesandbox.io/s/hopeful-hill-y44vp?file=/src/App.js

V6
https://codesandbox.io/s/priceless-visvesvaraya-42yd9?file=/src/App.js

**Expected behavior**
It should render the form when using ref property

**Additional context**
I have used the same kendo date picker with react-hook-form v6 and I have no issues. The thing I noticed is that in v6 the ref paramter is a ref object in v6. in v7 ref it is a callback ?

**Comment by abeprincec** [comment]
@bluebill1049  Is this solved in 7.11.1?

**Reply by bluebill1049** [reply]
you will need to wire those props, Controller no longer accesses input's value, either get or set. Controller only use ref for focus and native validation purpose.

**Reply by abeprincec** [reply]
We have to write our own onChange, ref property for the custom UI libraries ?

Sorry , I don't understand. Could you please provide an example or link to code example I can follow?

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api/usecontroller/controller

https://codesandbox.io/s/react-hook-form-v7-controller-5h1q5

**Reply by abeprincec** [reply]
I don't understand because my example is following the docs....

I do see some of the components have left of ref. If I leave out ref will it still focus on error?

**Reply by JeromeIllgner** [reply]
It seems to me that the docs for v7 are wrong. The docs for useController and controller say that the type of `ref` is `React.ref` but in fact it's a `RefCallBack`

Is there a way to get the actual ref from the ref callback?

**Reply by Moshyfawn** [reply]
[How to share a `ref` usage](https://react-hook-form.com/faqs/#Howtosharerefusage)

Docs are correct.

**Reply by Moshyfawn** [reply]
> If I leave out ref will it still focus on error?

No. In React, you **need** a `ref` to focus an input.

**Reply by JeromeIllgner** [reply]
Ahh yes,thank you for the clarification. TIL
